### DISCUSSION TITLE
Considering Kotlin interop

### DISCUSSION THREAD

**Discussion by SoyNegro** [lets-mica/mica#41] [Ideas]
Maybe a bit silly from me but after reading the code at [mica-mongo](https://github.com/lets-mica/mica/tree/master/mica-mongo) thought moving some small packages like that to Kotlin would be nice. Not only would make the code less verbose at points like [here](https://github.com/lets-mica/mica/blob/d3bd2b9f910e82c8b1818ed88c8d58ddea3ef538/mica-mongo/src/main/java/net/dreamlu/mica/mongo/utils/MongoJsonUtils.java#L111) where you can change the whole `else if` block for a `where` block. But you can also take advantage of Kotlin null safety going from `	public Document convert(@Nullable ObjectNode source) {
		return source == null ? null : Document.parse(source.toString());
	}` to `fun convert(ObjectNode source?):Document? = Document.parse(source?.toString()) `

**Comment by ChunMengLu** [comment]
I don't know much about Kotlin. As a basic component package, I feel that bundling a kotlin-stdlib is not friendly to java developers.
